### ISSUE TITLE
PDCurses-win10-jp

### ISSUE THREAD

**Issue by Hamayama** [wmcbrine/PDCurses#105]
Hello.

On Windows 10, the display corruption of Japanese characters occurs in wincon port.
(On Windows 8.1 or earlier, it doesn't occur.)

I made a modified version of PDCurses to fix this issue.
https://github.com/Hamayama/PDCurses-win10-jp
(Sorry, the document is Japanese only ...)

I'm using this as the front end of Lem editor.
https://github.com/cxxxr/lem

The whole difference to the mainstream could be see in the following URL.
https://github.com/Hamayama/PDCurses-win10-jp/compare/pdcurses-3.9-orig-618e0aa...master

I wonder this is too complex to merge to the mainstream.
(And also, MS is recently recommending VT escape sequence instead of Windows Console API.
So, this patch might be useless in few years ...)

So, this is just a information.

Related issues : #87 #92

**Comment by GitMensch** [comment]
> I wonder this is too complex to merge to the mainstream.

Definitely. I think the [array check fix](https://github.com/Bill-Gray/PDCursesMod/issues/199) should be applied in any case and I suggest to create a new PR for that.

> And also, MS is recently recommending VT escape sequence instead of Windows Console API.

In that case I highly suggest to check out the [VT port in PDCursesMod](https://github.com/Bill-Gray/PDCursesMod/tree/master/vt) and test how well that works for you in general.

**Comment by Bill-Gray** [comment]
The array check fixes definitely should be applied.  It appears to me that you have found [three places where array index checks are out of order](https://github.com/Bill-Gray/PDCursesMod/commit/c03e650a7020c8c2945d74fb4ecf258ff6d8e86f).

**Comment by Bill-Gray** [comment]
Actually,  a fourth check is in this pull release,  to ensure that `_new_packet()` does not attempt to handle more than 512 characters.  I would [suggest this fix to break oversized packets into a series of smaller packets](https://github.com/Bill-Gray/PDCursesMod/commit/51327cad4df8d4ea37e1ca188564074d4ae1a408).  For testing,  I tried  a `MAX_PACKET_SIZE` of 4,  just to ensure a situation in which lots of packets would be broken.  Speed suffered;  a 128-character packet size should avoid that problem.

Note that the X11 `PDC_transform_line()` has a similar issue,  amenable to a similar fix.  Possibly other platforms as well (my fork has no such limits in WinGUI and the SDLs.)  I'll push the X11 fix for my fork after checking other platforms.
